includes/class-newspack-ads-settings.php

```diff
@@ -0,0 +1,324 @@
+<?php
+/**
+ * Newspack Ads Settings
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack Ads Settings Class.
+ */
+class Newspack_Ads_Settings {
+
+	const API_NAMESPACE      = 'newspack-ads/v1';
+	const API_CAPABILITY     = 'manage_options';
+	const OPTION_NAME_PREFIX = '_newspack_ads_';
+
+	/**
+	 * Initialize settings.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints needed to fetch and update settings.
+	 */
+	public static function register_api_endpoints() {
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings-list',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_get_settings_list' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+			]
+		);
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_get_settings' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+				'args'                => [
+					'section' => [
+						'default'           => '',
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_update_section' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+				'args'                => [
+					'section'  => [
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
```

Comment:
The values for these settings are numbers—should this be `absint`? Although if we want to use this same endpoint for any arbitrary settings that could be text or numbers, I suppose `sanitize_text_field` works.

Reply:
In this specific case, it receives the section name, which is always a string. The settings data are sent through the `settings` arg and is sanitized [here](https://github.com/Automattic/newspack-ads/pull/213/files#diff-52802592e1cb377b15ce60d7ab90c75f6ae26f66be53709ea3c03b04aca7835fR104-R113), using `$config['type']` for typecasting, which I take suggestions on improvements!

Reply:
`absint` does not work for negative or float values. For lazy load specifically it is only an issue for the `mobile_scaling` setting, which is `float`, but in abstraction terms, it is not ideal for our settings.